### PR TITLE
FIX-670: more fixes

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
+ automatic patch.
+ some manual fixes

Well maybe not last :)

### CODE REVIEW

File: include/eve/detail/function/simd/ppc/swap_adjacent_groups.hpp [hunk 1]
```diff
@@ -13,11 +13,9 @@
...
 namespace eve::detail
 {
-  template<real_scalar_value T, typename N, ppc_abi ABI, std::ptrdiff_t G>
-  EVE_FORCEINLINE wide<T,N,ABI> swap_adjacent_groups_( EVE_SUPPORTS(cpu_)
-                                                    , wide<T,N,ABI> v, fixed<G>
-                                                    ) noexcept
-  requires(G<=N::value)
+  template<real_scalar_value T, typename N, std::ptrdiff_t G>
+  EVE_FORCEINLINE wide<T, N> swap_adjacent_groups_( EVE_SUPPORTS(vmx_), wide<T, N> v, fixed<G> ) noexcept
+    requires(G<=N::value)
```
Comment (jfalcou) [inline include/eve/detail/function/simd/ppc/swap_adjacent_groups.hpp:18]: missing ppc_abi check

File: include/eve/detail/function/simd/x86/arithmetic_compounds.hpp [hunk 2]
```diff
@@ -17,11 +17,11 @@ namespace eve::detail
   //================================================================================================
   // +=
   //================================================================================================
-  template<scalar_value T, value U, typename N, x86_abi ABI>
-  EVE_FORCEINLINE decltype(auto) self_add(wide<T, N, ABI> &self, U const &other) noexcept
-      requires(scalar_value<U> || std::same_as<wide<T, N, ABI>, U>)
+  template<scalar_value T, value U, typename N>
+  EVE_FORCEINLINE decltype(auto) self_add(wide<T, N> &self, U const &other) noexcept
+      requires(scalar_value<U> || std::same_as<wide<T, N>, U>) && x86_abi<abi_t<T, N>>
```
Comment (jfalcou) [inline include/eve/detail/function/simd/x86/arithmetic_compounds.hpp:22]: shouldn't it be in ( ) ?

File: include/eve/detail/function/simd/x86/arithmetic_compounds.hpp [hunk 3]
```diff
@@ -80,11 +80,11 @@ namespace eve::detail
   //================================================================================================
   // -=
   //================================================================================================
-  template<scalar_value T, value U, typename N, x86_abi ABI>
-  EVE_FORCEINLINE decltype(auto) self_sub(wide<T, N, ABI> &self, U const &other) noexcept
-      requires(scalar_value<U> || std::same_as<wide<T, N, ABI>, U>)
+  template<scalar_value T, value U, typename N>
+  EVE_FORCEINLINE decltype(auto) self_sub(wide<T, N> &self, U const &other) noexcept
+      requires(scalar_value<U> || std::same_as<wide<T, N>, U>) && x86_abi<abi_t<T, N>>
```
Comment (jfalcou) [inline include/eve/detail/function/simd/x86/arithmetic_compounds.hpp:85]: same remarks about ( ) the whole thing

File: include/eve/module/real/algorithm/function/regular/simd/x86/all.hpp [hunk 4]
```diff
@@ -1,51 +0,0 @@
-//==================================================================================================
-/**
-  EVE - Expressive Vector Engine
```
Comment (jfalcou) [inline include/eve/module/real/algorithm/function/regular/simd/x86/all.hpp:3]: why is it deleted ?

File: include/eve/module/real/core/function/numeric/simd/arm/neon/fms.hpp [hunk 5]
```diff
@@ -14,12 +14,12 @@
...
-  EVE_FORCEINLINE wide<T, N, ABI> fms_(EVE_SUPPORTS(neon128_),
+  template<decorator D, real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> fms_(EVE_SUPPORTS(neon128_),
                                        numeric_type const &,
-                                       wide<T, N, ABI> const &v0,
-                                       wide<T, N, ABI> const &v1,
-                                       wide<T, N, ABI> const &v2) noexcept
+                                       wide<T, N> const &v0,
+                                       wide<T, N> const &v1,
+                                       wide<T, N> const &v2) noexcept
```
Comment (jfalcou) [inline include/eve/module/real/core/function/numeric/simd/arm/neon/fms.hpp:22]: no requires on arm ABI ?

File: include/eve/module/real/core/function/numeric/simd/arm/neon/fnms.hpp [hunk 6]
```diff
@@ -14,12 +14,12 @@
...
-  EVE_FORCEINLINE wide<T, N, ABI> fnms_(EVE_SUPPORTS(neon128_),
+  template<decorator D, real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> fnms_(EVE_SUPPORTS(neon128_),
                                        numeric_type const &,
-                                       wide<T, N, ABI> const &v0,
-                                       wide<T, N, ABI> const &v1,
-                                       wide<T, N, ABI> const &v2) noexcept
+                                       wide<T, N> const &v0,
+                                       wide<T, N> const &v1,
+                                       wide<T, N> const &v2) noexcept
```
Comment (jfalcou) [inline include/eve/module/real/core/function/numeric/simd/arm/neon/fnms.hpp:22]: no requires on arm ABI ?

File: include/eve/module/real/core/function/pedantic/simd/arm/neon/fma.hpp [hunk 7]
```diff
@@ -14,12 +14,12 @@
...
-  EVE_FORCEINLINE wide<T, N, ABI> fma_(EVE_SUPPORTS(neon128_),
+  template<decorator D, real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> fma_(EVE_SUPPORTS(neon128_),
                                        pedantic_type const &,
-                                       wide<T, N, ABI> const &v0,
-                                       wide<T, N, ABI> const &v1,
-                                       wide<T, N, ABI> const &v2) noexcept
+                                       wide<T, N> const &v0,
+                                       wide<T, N> const &v1,
+                                       wide<T, N> const &v2) noexcept
```
Comment (jfalcou) [inline include/eve/module/real/core/function/pedantic/simd/arm/neon/fma.hpp:22]: no requires on arm ABI ?

File: include/eve/module/real/core/function/pedantic/simd/arm/neon/fms.hpp [hunk 8]
```diff
@@ -14,12 +14,12 @@
...
-  EVE_FORCEINLINE wide<T, N, ABI> fms_(EVE_SUPPORTS(neon128_),
+  template<real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> fms_(EVE_SUPPORTS(neon128_),
                                        pedantic_type const &,
-                                       wide<T, N, ABI> const &v0,
-                                       wide<T, N, ABI> const &v1,
-                                       wide<T, N, ABI> const &v2) noexcept
+                                       wide<T, N> const &v0,
+                                       wide<T, N> const &v1,
+                                       wide<T, N> const &v2) noexcept
```
Comment (jfalcou) [inline include/eve/module/real/core/function/pedantic/simd/arm/neon/fms.hpp:22]: no requires on arm ABI ?

File: include/eve/module/real/core/function/pedantic/simd/arm/neon/fnma.hpp [hunk 9]
```diff
@@ -14,12 +14,12 @@
...
-  EVE_FORCEINLINE wide<T, N, ABI> fnma_(EVE_SUPPORTS(neon128_),
+  template<real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> fnma_(EVE_SUPPORTS(neon128_),
                                        pedantic_type const &,
-                                       wide<T, N, ABI> const &v0,
-                                       wide<T, N, ABI> const &v1,
-                                       wide<T, N, ABI> const &v2) noexcept
+                                       wide<T, N> const &v0,
+                                       wide<T, N> const &v1,
+                                       wide<T, N> const &v2) noexcept
```
Comment (jfalcou) [inline include/eve/module/real/core/function/pedantic/simd/arm/neon/fnma.hpp:22]: no requires on arm ABI ?

File: include/eve/module/real/core/function/pedantic/simd/arm/neon/fnms.hpp [hunk 10]
```diff
@@ -14,12 +14,12 @@
...
-  template<real_scalar_value T, typename N, arm_abi ABI>
-  EVE_FORCEINLINE wide<T, N, ABI> fnms_(EVE_SUPPORTS(neon128_),
+  template<real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> fnms_(EVE_SUPPORTS(neon128_),
                                        pedantic_type const &,
-                                       wide<T, N, ABI> const &v0,
-                                       wide<T, N, ABI> const &v1,
-                                       wide<T, N, ABI> const &v2) noexcept
+                                       wide<T, N> const &v0,
+                                       wide<T, N> const &v1,
```
Comment (jfalcou) [inline include/eve/module/real/core/function/pedantic/simd/arm/neon/fnms.hpp:21]: no requires on arm ABI ?

File: include/eve/module/real/core/function/regular/simd/arm/neon/extract.hpp [hunk 11]
```diff
@@ -13,21 +13,20 @@
 
 namespace eve::detail
 {
-  template<typename T, typename N, typename I, auto V, arm_abi ABI>
+  template<typename T, typename N, typename I, auto V>
```
Comment (jfalcou) [inline include/eve/module/real/core/function/regular/simd/arm/neon/extract.hpp:16]: I need to check the usefulness of this function

File: include/eve/module/real/core/function/regular/simd/arm/neon/floor.hpp [hunk 12]
```diff
@@ -13,11 +13,11 @@
 
 namespace eve::detail
 {
-  template<floating_real_scalar_value T, typename N, arm_abi ABI>
-  EVE_FORCEINLINE wide<T, N, ABI> floor_(EVE_SUPPORTS(neon128_)
-                                       , wide<T, N, ABI> const& v) noexcept
+  template<floating_real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> floor_(EVE_SUPPORTS(neon128_)
+                                       , wide<T, N> const& v) noexcept
```
Comment (jfalcou) [inline include/eve/module/real/core/function/regular/simd/arm/neon/floor.hpp:18]: no requires on arm ABI ?

File: include/eve/module/real/core/function/regular/simd/arm/neon/minus.hpp [hunk 13]
```diff
@@ -14,10 +14,10 @@
 
 namespace eve::detail
 {
-  template<real_scalar_value T, typename N, arm_abi ABI>
-  EVE_FORCEINLINE wide<T, N, ABI> minus_(EVE_SUPPORTS(neon128_), wide<T, N, ABI> const& v) noexcept
+  template<real_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T, N> minus_(EVE_SUPPORTS(neon128_), wide<T, N> const& v) noexcept
```
Comment (jfalcou) [inline include/eve/module/real/core/function/regular/simd/arm/neon/minus.hpp:18]: no requires on arm ABI ?

File: include/eve/concept/abi.hpp [hunk 14]
```diff
@@ -19,35 +19,16 @@ namespace eve
     template<typename Wide, typename... ABI>
     struct  is_native_for_abi
           : std::bool_constant
-            < ( std::is_same_v< Wide
-                              , wide< element_type_t<Wide>
-                                    , expected_cardinal_t<element_type_t<Wide>, ABI>
-                                    , ABI
-                                    >
-                              >
+            < ( (std::is_same_v<typename Wide::abi_type, ABI> &&
```
Comment (jfalcou) [inline include/eve/concept/abi.hpp:22]: Is it enough ?

File: include/eve/module/real/algorithm/function/regular/simd/x86/all.hpp (hunk 4 above)
Comment (DenisYaroshevskiy) [reply include/eve/module/real/algorithm/function/regular/simd/x86/all.hpp:3]: It's dead and I know for a fact it's dead.

File: include/eve/concept/abi.hpp (hunk 14 above)
Comment (DenisYaroshevskiy) [reply include/eve/concept/abi.hpp:22]: The tests pass

File: include/eve/module/real/algorithm/function/regular/simd/x86/all.hpp (hunk 4 above)
Comment (jfalcou) [reply include/eve/module/real/algorithm/function/regular/simd/x86/all.hpp:3]: Good

**Review by jfalcou — APPROVED** [review]

**Pull request merged**